app/Console/Commands/InitCommand.php

```diff
@@ -237,8 +250,22 @@ private function maybeSetUpDatabase(): void
                 break;
             } catch (Throwable $e) {
                 $this->error($e->getMessage());
-                $this->warn(PHP_EOL . "Koel cannot connect to the database. Let's set it up.");
-                $this->setUpDatabase();
+
+                // We only try update credentials if running in interactive mode.
+                // Otherwise, we require admin intervention to fix them.
+                // This avoids inadvertently wiping credentials if there's a connection failure.
```

Comment:
Sorry, I missed this earlier, but how can credentials be wiped if there's a connection failure?

Reply:
If the connection fails, it'll throw an exception. So that then puts us here. We call setUpDatabase, which prompts for new credentials. Once it's done that, it writes them back to the env configuration.

This isn't all that bad, except that when not interactive, there is nobody to prompt for new credentials. So you end up writing empty strings back to the env file, overwriting the (probably good) settings that were there before.

If you want to test what happens yourself, set Koel up, then shut it down. Deliberately break your database connectivity somehow (perhaps renaming it, temporarily shutting it down, whatever - the simplest option I have used is to alter the settings in my env file to deliberately be incorrect) - and then try launch Koel in non-interactive mode.

You'll get endless terminal spam, and your env file will have the variables overwritten.

Reply:
> This isn't all that bad, except that when not interactive, there is nobody to prompt for new credentials. So you end up writing empty strings back to the env file, overwriting the (probably good) settings that were there before.

Credentials are not written back into `.env` file, but saved into the database, which can't be done since the DB connection is broken atm, no?

Reply:
Now I'm confused: surely, the credentials to _access_ the database cannot possibly be stored in the database that the credentials access...? Perhaps we're talking about different things? I suggest taking a look at the log files I attached, that demonstrate the problem (and that the patch fixes it).

I'm talking about the DB_* variables, which are stored in .env, and written back to the .env file by the following loop in setUpDatabase:

```
        foreach ($config as $key => $value) {
            $this->dotenvEditor->setKey($key, $value);
        }
```

Reply:
Ah, you mean the database credentials, not the admin. That makes sense now, thanks.